components/metadata-service/internal/metadata/secrets/service.go

```diff
@@ -0,0 +1,282 @@
+package secrets
+
+import (
+	"github.com/kyma-project/kyma/components/metadata-service/internal/apperrors"
+	"github.com/kyma-project/kyma/components/metadata-service/internal/k8sconsts"
+	"github.com/kyma-project/kyma/components/metadata-service/internal/metadata/model"
+	"github.com/kyma-project/kyma/components/metadata-service/internal/metadata/remoteenv"
+)
+
+const (
+	OauthClientIDKey     = "clientId"
+	OauthClientSecretKey = "clientSecret"
+
+	BasicAuthUsernameKey = "username"
+	BasicAuthPasswordKey = "password"
+)
+
+type Service interface {
+	Get(remoteEnvironment string, credentials remoteenv.Credentials) (model.Credentials, apperrors.AppError)
+	Create(remoteEnvironment, serviceID string, credentials *model.Credentials) (remoteenv.Credentials, apperrors.AppError)
+	Update(remoteEnvironment, serviceID string, credentials *model.Credentials) (remoteenv.Credentials, apperrors.AppError)
+	Delete(name string) apperrors.AppError
+}
+
+type service struct {
+	nameResolver k8sconsts.NameResolver
+	repository   Repository
+}
+
+func NewService(repository Repository, nameResolver k8sconsts.NameResolver) Service {
+	return &service{
+		nameResolver: nameResolver,
+		repository:   repository,
+	}
+}
+
+func (s *service) Create(remoteEnvironment, serviceID string, credentials *model.Credentials) (remoteenv.Credentials, apperrors.AppError) {
+	remoteEnvCredentials := remoteenv.Credentials{}
+
+	if credentials == nil {
+		return remoteEnvCredentials, nil
+	}
+
+	if s.basicCredentialsProvided(credentials) && s.oauthCredentialsProvided(credentials) {
+		return remoteEnvCredentials, apperrors.WrongInput("Creating access service failed: Multiple authentication methods provided.")
+	}
+
+	name := s.nameResolver.GetResourceName(remoteEnvironment, serviceID)
+
+	err := s.createCredentialsSecret(remoteEnvironment, serviceID, name, credentials)
+	if err != nil {
+		return remoteEnvCredentials, err
+	}
+
+	if s.oauthCredentialsProvided(credentials) {
```

Comment:
You could add a function converting model.Credentials to remoteEnv.Credentials and reuse it in Update method.